src/components/MultiHopTrade/components/TradeInput/components/HighlightedTokensCategoryDialog.tsx

```diff
@@ -0,0 +1,69 @@
+import { CheckIcon } from '@chakra-ui/icons'
+import { Box, Icon, Menu, MenuItemOption, MenuOptionGroup } from '@chakra-ui/react'
+import { useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+
+import { Dialog } from '@/components/Modal/components/Dialog'
+import { MarketsCategories } from '@/pages/Markets/constants'
+
+type HighlightedTokensCategoryDialogProps = {
+  isOpen: boolean
+  onClose: () => void
+  selectedCategory: MarketsCategories
+  handleCategoryChange: (category: MarketsCategories) => void
+}
+
+const checkedIcon = <Icon as={CheckIcon} color='blue.200' fontSize='20px' />
+
+export const HighlightedTokensCategoryDialog = ({
+  isOpen,
+  onClose,
+  selectedCategory,
+  handleCategoryChange,
+}: HighlightedTokensCategoryDialogProps) => {
+  const translate = useTranslate()
+
+  const categories = useMemo(() => {
```

Comment:
stylistic: feels weird - the JSX for a single category could be extracted as `<Category />`  then `categories would simply be 

```typescript
Object.values(MarketsCategories).map(category => {
  // some logic here
  return <Category />
}
```

Reply:
Changed in [6a578a1](https://github.com/shapeshift/web/pull/10100/commits/6a578a1a0cf63fe1f2f686588e99f43f24846752)